Adding a category to the link object to support multiple link types.

DavidEBest opened the pull request:
As per 18F/dashboard#266, there should be a way to specify an API link in the about.yml file. This duplicates the structure of the links object to provide more flexibility.

mbland commented:
This looks good to me, but I'll wait to see if other folks have any comments before merging.

ertzeid commented:
Looks good to me.

mbland commented:
Actually, going to leave this open for now until #40 is resolved.

arowla commented:
Personally, I'd like to better understand why API links need a separate section vs. changing the links section to be able to tag or label links, like so:

``` yaml
links:
  description: api
  url: https://api.data.gov/foo
```

DavidEBest commented:
Actually, @arowla, that came up in our dashboard sprint meeting. I'll be committing a change momentarily to do exactly that. :)

DavidEBest commented:
As per the dashboard sprint planning conversation, I've removed the specific `api_link` field and added a `category` to the link type.

`category` is an enumerated string that accepts `api`, `process`, or `related`. I chose the name `category` because we use it for the `services` field.

arowla commented:
Wonderful. :) Could you elaborate on what 'process' stands for?

DavidEBest commented:
The inspiration for the `process` category is https://github.com/18F/dashboard/issues/265, which wanted a section for links to waffle, trello, etc. I thought `process` encapsulated that idea decently. Open to suggestions, though.

arowla commented:
Ok! That was my guess, but wanted to be sure. Thanks!

mbland commented:
Again, looks good, per our discussion earlier. Would like @ccostino, @ertzeid, @annalee, @melodykramer, @jeremiak and @gboone, to also have a chance to comment before merging. Barring objections or further suggestions, I'll merge tomorrow morning.

ertzeid commented:
LGTM

ccostino commented:
Adding my +1 as well, this looks fine to me!

gboone commented:
Looking good. One question about formatting for the about.yml files, will they support a `text` field to fill in the text that would display inside an anchor tag?

Also, is this correct formatting for multiple links?

```
links:
  - description: api
    url: https://team-api.18f.gov
  - description: process
    url: https://trello.com/some-project-specific-string
```

mbland commented:
> Looking good. One question about formatting for the about.yml files, will they support a text field to fill in the text that would display inside an anchor tag?

They already do. It's the entry in the schema immediately before this one.

> Also, is this correct formatting for multiple links?

Looks like it would be this (note `category` instead of `description`):

``` yaml
links:
- text: Team API server
  category: api
  url: https://team-api.18f.gov
- text: Task tracking (Trello)
  category: process
  url: https://trello.com/some-project-specific-string
- text: 18F home page
  category: related
  url: https://18f.gsa.gov/
```

DavidEBest commented:
@gboone The link structure contains `url`, `text`, and `category`. The way to specify multiple would be:

```
links:
- url: https://url...
  text: Related link 1
  category: related
- url: ...
  text: Related Link 2
- url: ...
  text: API Link 1
  category: api
```

@mbland Should the category property be marked as optional? Or defaulted to a specific value (assuming that is possible in JSON Schema)?

mbland commented:
@DavidEBest Was wondering about that. I'm thinking optional, but am open to other ideas.

gboone commented:
Cool, thanks for the clarification @DavidEBest.
On the default or optional question: For 18F/dashboard `related` would be a good default, otherwise I'd vote for making it optional.

DavidEBest commented:
Yeah, like Mike, I'm leaning toward optional. Until we get versioning in place, we should be careful about adding new required attributes. I'll add an optional attribute to the category field as soon as I figure out how. :)

DavidEBest commented:
So, @mbland, all three attributes are currently optional. Is that okay, or should I add a required attribute for the url?

mbland commented:
@DavidEBest Sure, requiring the URL seems like the right thing to do.
